src/mark.d.ts

```diff
@@ -0,0 +1,79 @@
+import type {Dimensions} from "./dimensions.js";
+import type {ChannelDomainSort, ChannelSpec, ChannelValue, ChannelValueSpec} from "./channel.js";
+import type {Context} from "./context.js";
+import type {Facet, FacetAnchor} from "./facet.js";
+import type {plot} from "./plot.js";
+
+export type Data = Iterable<any> | {length: number};
+
+export type RenderFunction = (
+  index: number[],
+  scales: {[name: string]: (value: any) => any},
+  values: {[name: string]: any[]},
```

Comment:
channels?

Reply:
Yes these are derived from the mark’s declared channels.